packages/core/server/admin-api/services/query-layer-decorator.ts

```diff
@@ -67,20 +67,26 @@ const decorator = (service: IDecoratedService) => ({
       return service.update.call(this, uid, entityId, opts);
     }
 
+    // Manually fetch the entity that's being updated.
+    // We do this becuase not all it's data is present in opts.data.
+    const entity = await service.findOne.call(this, uid, entityId, {
+      populate: {
+        url_alias: {
+          fields: ['id'],
+        },
+      },
+    });
+
     // If a URL alias is allready present, fetch it.
-    if (opts.data.url_alias) {
-      urlAliasEntity = await getPluginService('urlAliasService').findOne(opts.data.url_alias);
+    if (opts.data.url_alias || entity.url_alias?.id) {
+      urlAliasEntity = await getPluginService('urlAliasService').findOne(opts.data.url_alias || entity.url_alias?.id);
```

Comment:
We don't need to refetch the entity if we just populated it

```suggestion
    if (opts.data.url_alias) {
      urlAliasEntity = await getPluginService('urlAliasService').findOne(opts.data.url_alias);
    } else if (entity.url_alias) {
      urlAliasEntity = entity.url_alias;
    }
```

For this to work, you should update the `opts` type to

```ts
    opts: IDecoratedServiceOptions<{ url_alias: Entity.ID }>,
```

Reply:
Done